src/L1/rollup/ScrollChain.sol

```diff
@@ -575,12 +663,151 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
+            abi.encodePacked(_blobVersionedHash, _blobDataProof)
+        );
+        // We verify that the point evaluation precompile call was successful by testing the latter 32 bytes of the
+        // response is equal to BLS_MODULUS as defined in https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile
+        if (!success) revert ErrorCallPointEvaluationPrecompileFailed();
+        (, uint256 result) = abi.decode(data, (uint256, uint256));
+        if (result != BLS_MODULUS) revert ErrorUnexpectedPointEvaluationPrecompileOutput();
+    }
+
+    /// @dev Internal function to check the `SkippedL1MessageBitmap`.
+    /// @param _totalL1MessagesPoppedOverall The total number of L1 messages popped after current batch.
+    /// @param _totalL1MessagesPoppedInBatch The total number of L1 messages popped in current batch.
+    /// @param _skippedL1MessageBitmap The skipped L1 message bitmap in calldata.
+    /// @param _doPopMessage Whether we actually pop the messages from message queue.
+    function _checkSkippedL1MessageBitmap(
+        uint256 _totalL1MessagesPoppedOverall,
+        uint256 _totalL1MessagesPoppedInBatch,
+        bytes calldata _skippedL1MessageBitmap,
+        bool _doPopMessage
+    ) private {
+        // check the length of bitmap
+        unchecked {
+            if (((_totalL1MessagesPoppedInBatch + 255) / 256) * 32 != _skippedL1MessageBitmap.length) {
+                revert ErrorIncorrectBitmapLength();
+            }
+        }
+        if (_doPopMessage) {
+            _popL1MessagesCalldata(
+                _skippedL1MessageBitmap,
+                _totalL1MessagesPoppedOverall,
+                _totalL1MessagesPoppedInBatch
+            );
+        }
+    }
+
+    /// @dev Internal function to get the blob versioned hash.
+    /// @return _blobVersionedHash The retrieved blob versioned hash.
+    function _getBlobVersionedHash() internal virtual returns (bytes32 _blobVersionedHash) {
+        bytes32 _secondBlob;
+        // Get blob's versioned hash
+        assembly {
+            _blobVersionedHash := blobhash(0)
+            _secondBlob := blobhash(1)
+        }
+        if (_blobVersionedHash == bytes32(0)) revert ErrorNoBlobFound();
+        if (_secondBlob != bytes32(0)) revert ErrorFoundMultipleBlobs();
+    }
+
     /// @dev Internal function to commit chunks with version 0
     /// @param _totalL1MessagesPoppedOverall The number of L1 messages popped before the list of chunks.
     /// @param _chunks The list of chunks to commit.
     /// @param _skippedL1MessageBitmap The bitmap indicates whether each L1 message is skipped or not.
     /// @return _batchDataHash The computed data hash for the list of chunks.
-    /// @return _totalL1MessagesPoppedInBatch The total number of L1 messages poped in this batch, including skipped one.
+    /// @return _totalL1MessagesPoppedInBatch The total number of L1 messages popped in this batch, including skipped one.
```

Comment:
since this pr tends to reduce gas in the meantime, come with a question, could we remove `maxNumTxInChunk` check in this contract?

Reply:
because updating this value may need a timelock.

Reply:
we don't limit `maxNumTxInChunk` in circuit any more?

Reply:
still limit `maxNumTxInChunk`, just wondering if this check can only be done in verifier.

Reply:
verifier will fail to verify and then we need to revert batch.
early stop in commit always better.

Reply:
makes sense. and the upgrades when increasing this value could be async.